src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -55,10 +55,30 @@
   <UsingTask AssemblyFile="$(XPlatSourceBuildTasksAssembly)" TaskName="WritePackageUsageData" />
   <UsingTask AssemblyFile="$(XPlatSourceBuildTasksAssembly)" TaskName="WriteUsageReports" />
 
-  <!-- Update NuGet feeds in the repo -->
+  <Target Name="CopyNuGetConfig"
+          BeforeTargets="
+            UpdateNuGetConfig;
+            Build"
+          Condition="'$(NuGetConfigFile)' != ''"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(BaseIntermediateOutputPath)CopyNuGetConfig.complete">
```

Comment:
We don't need a semaphore file here. The output that the target produces is `NuGetConfigFile`.

Reply:
Fixed with https://github.com/dotnet/installer/pull/18478/commits/27b92a3eee2f070b326230688f4dad59299358cd